src/cpu-template-helper/src/main.rs

```diff
@@ -89,3 +89,60 @@ fn main() {
         std::process::exit(EXIT_CODE_ERROR);
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::io::Write;
+
+    use ::utils::tempfile::TempFile;
+    use vmm::utilities::mock_resources::kernel_image_path;
+
+    use super::*;
+
+    pub fn generate_config(kernel_image_path: &str, rootfs_path: &str) -> String {
+        format!(
+            r#"{{
+                "boot-source": {{
+                    "kernel_image_path": "{}"
+                }},
+                "drives": [
+                    {{
+                        "drive_id": "rootfs",
+                        "path_on_host": "{}",
+                        "is_root_device": true,
+                        "is_read_only": false
+                    }}
+                ]
+            }}"#,
+            kernel_image_path, rootfs_path,
+        )
+    }
+
+    fn generate_valid_config_file(kernel_image_path: &str, rootfs_path: &str) -> TempFile {
+        let config = generate_config(kernel_image_path, rootfs_path);
+        let config_file = TempFile::new().unwrap();
+        config_file.as_file().write_all(config.as_bytes()).unwrap();
+        config_file
+    }
+
+    #[test]
+    fn test_dump_command() {
+        let kernel_image_path = kernel_image_path(None);
+        let rootfs_file = TempFile::new().unwrap();
+        let config_file =
+            generate_valid_config_file(&kernel_image_path, rootfs_file.as_path().to_str().unwrap());
+        let output_file = TempFile::new().unwrap();
+
+        let args = vec![
+            "cpu-template-helper",
+            "dump",
+            "--config",
+            config_file.as_path().to_str().unwrap(),
+            "--output",
+            output_file.as_path().to_str().unwrap(),
```

Comment:
Could the command just output to stdout? This allows for both outputting to a file e.g. `cpu-helper-tool dump >> cpu_template_dump` and to stdout for checking and testing.

Reply:
Yes, it could. The functionality of dump feature is out of scope of this PR. Will change the dump command to output to stdout when `--output` is not provided in a follow-up PR.

Reply:
Created a task for this. Closing this conversation for now.